### PR TITLE
CLN: move code out of a try clause in merge.py

### PR DESCRIPTION
**Pull request by topper-123**
Minor clean-up.

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -116,20 +116,20 @@ def _groupby_and_merge(
...
-                right = _right
+    # we will check & remove duplicates if indicated
+    if check_duplicates:
+        if on is None:
+            on = []
+        elif not isinstance(on, (list, tuple)):
+            on = [on]
+
+        if right.duplicated(by + on).any():
+            _right = right.drop_duplicates(by + on, keep="last")
```
Comment (jbrockmendel) [inline pandas/core/reshape/merge.py]: no risk of raising in 127/128?

Comment (topper-123) [reply pandas/core/reshape/merge.py]: It was a risk already, right? This just ensures that a KeyError isn't catched unintended.

Comment (jbrockmendel) [reply pandas/core/reshape/merge.py]: makes sense, thanks

**Comment by jbrockmendel** [comment]
Thanks @topper-123

**Pull request merged**